doc: update Reviewing section of onboarding doc

Trott opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
doc

##### Description of change
<!-- Provide a description of the change below this comment. -->

addaleax commented:
Since this seems like a good place to voice suggestions regarding the process…
I think it might be a good idea to write out “Looks good to me” at least the first one or two times for first-time contributors. Not everyone knows the abbreviation, and while it is easily looked up, it might help with making Node core seem like less of an insider’s club?

addaleax commented on doc/onboarding.md:
```diff
@@ -64,31 +64,29 @@ onboarding session.
-    * primary goal is for the codebase to improve
-    * secondary (but not far off) is for the person submitting code to succeed
-      * helps grow the community
-      * and draws new people into the project
```
Did you feel these two points were redundant/any other specific reason to drop them?

Fishrock123 replied:
sadly I'm not sure everyone understands this by default so distilling the why is probably still useful imo

Trott replied:
I was trying to decide which line to remove because they seemed to be saying the same thing just using different words.

Then I thought the document is probably fine without either of them and removed them both.

I think what those two lines say are implied by the line above them.

But I don't feel that strongly about it and can restore one or the other if anyone feels strongly about it.

Trott commented:
💯 on writing out `LGTM = Looks Good To Me`. It's also why I defined `nit` above.

Trott commented:
@addaleax @Fishrock123 

* Spelled out what `LGTM` stands for
* Added a sentence explaining why we want new people to be successful (restoring the "grow the community" wording).
* Similar to defining `LGTM` and `nit`, I also replaced all uses of `PR` in this section with `pull request`. (If we like that, I can do it throughout the doc. If we'd rather keep it because people might see `PR` elsewhere, I can explain the acronym in another section. Happy to punt on that for now. I plan on treating this document as more of a living document than I have been, so I don't feel the need to cram in every improvement into this PR...er...this pull request. There will be others...)

addaleax commented:
LGTM ;)

targos commented:
Looks good to me

fhinkel commented:
LGTM

evanlucas commented:
LGTM

jasnell commented:
LGTM

Trott commented:
Landed in b180a5beca7c13c96342c97d216fc85673c815b9
